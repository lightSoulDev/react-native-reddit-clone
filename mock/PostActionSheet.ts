export const PostActionSheet = [
  {
    label: 'Share',
    onPress: () => {
      console.log('Share');
    },
    materialIcon: 'share',
  },
  {
    label: 'Save',
    onPress: () => {
      console.log('Save');
    },
    materialIcon: 'archive',
  },
  {
    label: 'Copy text',
    onPress: () => {
      console.log('Copy text');
    },
    materialIcon: 'content-copy',
  },
  {
    label: 'Report',
    onPress: () => {
      console.log('Report');
    },
    materialIcon: 'alert-octagon',
  },
  {
    label: 'Block user',
    onPress: () => {
      console.log('Block user');
    },
    materialIcon: 'account-cancel',
  },
  {
    label: 'Hide',
    onPress: () => {
      console.log('Hide');
    },
    materialIcon: 'eye-off',
  },
];
