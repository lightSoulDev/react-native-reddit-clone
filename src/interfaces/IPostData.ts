import {IAwardData} from './IAwardData';
import {ICommentData} from './ICommentData';

export interface IPostData {
  avatarIcon: any;
  community: string;
  author: string;
  publishDate: number;
  upvotes: number;
  extras?: string;
  comments?: Array<ICommentData>;
  awards?: Array<IAwardData>;
}
