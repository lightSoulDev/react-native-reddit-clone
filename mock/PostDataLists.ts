import {IPostData} from '../src/interfaces/IPostData';

export const homeScreenPosts: Array<IPostData> = [
  {
    id: 'p_0',
    avatarIcon: null,
    community: 'react-native',
    author: 'lightSoulDev',
    extras: 'i.reddit.io',
    publishDate: 1647146918 * 1000,
    upvotes: 100,
  },
];

export const newsScreenPosts: Array<IPostData> = [
  {
    id: 'n_1',
    avatarIcon: null,
    community: 'react-js',
    author: 'lightSoulDev',
    extras: 'i.reddit.io',
    publishDate: 1647060518 * 1000,
    upvotes: 40,
  },
];
