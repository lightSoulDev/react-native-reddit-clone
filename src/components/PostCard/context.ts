import React, {useContext} from 'react';
import {IPostData} from '../../interfaces/IPostData';

interface ContextState {
  postData: IPostData;
  setPostData: React.Dispatch<React.SetStateAction<IPostData>> | (() => void);
}

export const PostContext = React.createContext<ContextState>({
  postData: {} as IPostData,
  setPostData: () => {},
});

export const usePostContext = () => useContext(PostContext);
