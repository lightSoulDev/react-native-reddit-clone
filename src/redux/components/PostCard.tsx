import {View, Text, Image, StyleSheet, TouchableOpacity} from 'react-native';
import React from 'react';
import {styles} from './PostCardStyles';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {useSheet} from '../../context/ActionSheetContext';
const MockCommunityIcon = require('./../../assets/mock/community_icon.png');

export interface PostCardProps {
  backgroundColor: string;
}

const PostCardHeader = (props: any) => {
  const {toggleSheet} = useSheet();

  return (
    <View style={hstyles(props).container}>
      <TouchableOpacity>
        <Image style={hstyles(props).icon} source={MockCommunityIcon} />
      </TouchableOpacity>
      <View>
        <TouchableOpacity>
          <Text>r/react-native</Text>
        </TouchableOpacity>
        <View style={hstyles(props).container}>
          <TouchableOpacity>
            <Text>u/lightSoulDev</Text>
          </TouchableOpacity>
          <Text> - </Text>
          <Text>15h</Text>
          <Text> - </Text>
          <TouchableOpacity>
            <Text> i.reddit.io </Text>
          </TouchableOpacity>
        </View>
      </View>
      <TouchableOpacity
        style={hstyles(props).options}
        onPress={() => {
          toggleSheet([
            {
              label: 'Some Label',
              onPress: () => {
                console.log('Some Label');
              },
              materialIcon: '',
            },
            {
              label: 'Some Label 1',
              onPress: () => {
                console.log('Some Label');
              },
              materialIcon: '',
            },
            {
              label: 'Some Label 2',
              onPress: () => {
                console.log('Some Label');
              },
              materialIcon: '',
            },
            {
              label: 'Some Label 3',
              onPress: () => {
                console.log('Some Label');
              },
              materialIcon: '',
            },
          ]);
        }}>
        <MaterialCommunityIcons name="dots-horizontal" size={25} />
      </TouchableOpacity>
    </View>
  );
};

const hstyles = (props: any) =>
  StyleSheet.create({
    options: {
      position: 'absolute',
      right: 10,
      alignSelf: 'center',
    },
    headerLinks: {
      flexDirection: 'column',
    },
    container: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    icon: {
      width: 40,
      height: 40,
      marginRight: 10,
      borderRadius: 40,
    },
  });

const PostCard = (props: PostCardProps) => {
  const {backgroundColor} = props;

  return (
    <View style={styles(props).container}>
      <PostCardHeader />
    </View>
  );
};

export default PostCard;
