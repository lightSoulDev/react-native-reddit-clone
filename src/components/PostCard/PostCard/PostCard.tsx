import {View} from 'react-native';
import React from 'react';
import {styles} from './styles';
import PostCardHeader from '../PostCardHeader/PostCardHeader';
import PostCardFooter from '../PostCardFooter/PostCardFooter';
import {IPostData} from '../../../interfaces/IPostData';
import {PostContext} from '../context';

export interface PostCardProps {
  backgroundColor: string;
  postData: IPostData;
}

const PostCard = (props: PostCardProps) => {
  return (
    <PostContext.Provider value={props.postData}>
      <View style={styles(props).container}>
        <PostCardHeader />
        {/* Awards */}
        {/* Tags */}
        <View style={{height: 300, backgroundColor: '#ccc'}}></View>
        <PostCardFooter />
      </View>
    </PostContext.Provider>
  );
};

export default PostCard;
