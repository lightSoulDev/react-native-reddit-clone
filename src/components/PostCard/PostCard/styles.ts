import {StyleSheet} from 'react-native';
import {PostCardProps} from './PostCard';

export const styles = (props: PostCardProps) =>
  StyleSheet.create({
    container: {
      width: '100%',
      padding: 12,
      marginVertical: 5,
      backgroundColor: props.backgroundColor,
    },
  });
