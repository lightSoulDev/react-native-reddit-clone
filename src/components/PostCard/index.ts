import PostCard from './PostCard/PostCard';

export default PostCard;
