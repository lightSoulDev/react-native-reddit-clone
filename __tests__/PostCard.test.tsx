import {NativeBaseProvider, View} from 'native-base';
import React, {useState} from 'react';
import {create, act} from 'react-test-renderer';
import {allPosts} from '../mock/PostDataLists';
import PostCard from '../src/components/PostCard';
import {IPostData} from '../src/interfaces/IPostData';

const Container = () => {
  const [postDataArray, setPostDataArray] = useState<IPostData[]>([]);

  return <NativeBaseProvider>{post}</NativeBaseProvider>;
};

const post = create(<PostCard backgroundColor="#aaa" postData={allPosts[0]} />);
const tree = create(<Container />);

test('snapshot', () => {
  expect(post).toMatchSnapshot();
});

// TODO: Upvotes test
