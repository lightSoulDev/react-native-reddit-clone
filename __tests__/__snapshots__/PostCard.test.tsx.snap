// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#aaa",
      "marginVertical": 5,
      "padding": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 40,
            "height": 40,
            "marginRight": 10,
            "width": 40,
          }
        }
      />
    </View>
    <View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text>
          r/react-native
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            u/lightSoulDev
          </Text>
        </View>
        <Text>
           - 
        </Text>
        <Text>
          12 Hr
        </Text>
        <Text>
           - 
        </Text>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
             i.reddit.io 
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
          "position": "absolute",
          "right": 10,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ccc",
        "height": 300,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "left": 5,
          "position": "absolute",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
        <Text>
          100
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text />
        <Text>
           Comment
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "position": "absolute",
          "right": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "opacity": 1,
          }
        }
      >
        <Text />
        <Text>
           Share
        </Text>
      </View>
    </View>
  </View>
</View>
`;
